src/v/cluster/health_monitor_frontend.cc

```diff
@@ -147,4 +147,25 @@ ss::future<bool> health_monitor_frontend::does_raft0_have_leader() {
       [](health_monitor_backend& be) { return be.does_raft0_have_leader(); });
 }
 
+ss::future<> health_monitor_frontend::refresh_info() {
+    vlog(clusterlog.info, "Refreshing disk health info");
+    // Force a refresh to update internal state on the backend.
+    auto deadline = model::time_from_now(default_timeout);
+    return dispatch_to_backend([deadline](health_monitor_backend& be) {
+               return be.get_cluster_disk_health(force_refresh::yes, deadline);
+           })
+      .then([this](storage::disk_space_alert dsa) {
+          if (dsa != _cluster_disk_health) {
+              vlog(
+                clusterlog.debug,
+                "Update disk health cache {} -> {}",
+                _cluster_disk_health,
+                dsa);
+              _cluster_disk_health = dsa;
+              return update_other_shards(dsa);
```

Comment:
it looks like `update_disk_health_cache` is run via a timer `disk_health_tick`. if these methods are functionally the same, and now that functionality races with each other, perhaps we stop stop that timer, wait for on going refresh to stop, then do the on demand refresh and reset the timer?

Reply:
Resolved offline with Michal and Noah. This was the result of miss-communication.
The requested change was to rename `update_disk_health_cache` to something more descriptive.